src/runtime/nitro/plugins/40-cspSsrNonce.ts

```diff
@@ -2,9 +2,9 @@ import { defineNitroPlugin } from '#imports'
 import crypto from 'node:crypto'
 import { resolveSecurityRules } from '../context'
 
-const LINK_RE = /<link([^>]*?>)/g
-const SCRIPT_RE = /<script([^>]*?>)/g
-const STYLE_RE = /<style([^>]*?>)/g
+const LINK_RE = /<link([^>]*?>)/gi
```

Comment:
Can links also ne uppercase? And more importantly, do links pose a security vulnerability enough to warrant a case insensitive search, if at all? For scripts it's understandable,  it can have detrimental consequences, but links and styles?

Reply:
I agree CodeQL sounds a bit excessive.
However if we don't add the nonce to an uppercase `<LINK>` tag, the page will be blocked. I'd say better be on the safe side.

Reply:
I totally agree that we should err on the side of caution, but the question is whether the caution is warranted. Is there confirmation that upper case link tags get parsed correctly by browsers and that they have any effect? If these tags don't do anything anyways, there's no benefit from making a more expensive pattern to capture them.

Reply:
HTML W3C Standard : https://html.spec.whatwg.org/multipage/syntax.html#elements-2

Reply:
Thanks for the resource! I'm going to silently judge W3C for this standard and the recipe they're making for a disaster, but if that's the case then yea, the I flag is necessary.

Reply:
😂👍🏻